source/_UIAHandler.py

```diff
@@ -224,9 +230,40 @@ def MTAThreadFunc(self):
 			self.MTAThreadInitException=e
 		finally:
 			self.MTAThreadInitEvent.set()
-		self.MTAThreadStopEvent.wait()
+		while True:
+			func=self.MTAThreadQueue.get()
+			if func:
+				func()
+			else:
+				break
 		self.clientObject.RemoveAllEventHandlers()
 
+	def _onForegroundChange(self):
+		pendingForegroundUIAElement=self.pendingForegroundUIAElement
+		if pendingForegroundUIAElement==self.currentForegroundUIAElement:
+			return
+		if self.currentForegroundUIAElement:
+			try:
+				self.clientObject.removePropertyChangedEventHandler(self.currentForegroundUIAElement,self)
+			except COMError:
+				# The old UIAElement died as the window was closed.
+				# The system should forget the old event registration itself.
+				pass
+		self.currentForegroundUIAElement=pendingForegroundUIAElement
+		try:

```

Comment:
Is event registration required if not handler.isNativeUIAElement(pendingForegroundUIAElement)?

Reply:
Yes. UIA can be implemented per-window. Just because the foreground window does not support UIA, does not mean that one of its descendants does not.